### DISCUSSION TITLE
Why doesn't Kratos continue the Hydra OAuth2 flow after login with `login_challenge`?

### DISCUSSION THREAD

**Discussion by davidknezic** [ory/kratos#3063] [Q&A]
I'm struggling to find good examples for the recently added Kratos + Hydra integration.

As far as I understood, all that's really needed is the `OAUTH2_PROVIDER_URL` env var to be set to the Hydra admin url (e.g. http://localhost:4445).

With that set, I extended the [Kratos Next.js self-service UI example](https://github.com/ory/kratos-selfservice-ui-react-nextjs/blob/master/pages/login.tsx) so that the `login.tsx` page accepts the `login_challenge` query param and passes it on to the `createBrowserLoginFlow`.

This seems to work so far, because I notice that Kratos now initiates a login request with Hydra, which is very cool! Also, there's the `oauth2_login_challenge` set on my flow payload.

However, I notice that the `oauth2_login_request` field isn't present at all on the flow. Shouldn't this be the case, though?

Now I can continue to log in through my browser. However, when I complete the login flow, I receive a `HTTP 200` response that contains the newly issued session.

My question now is, how do I get Kratos to continue the OAuth2 flow? I'd expect it to request a redirect to a public Kratos or Hydra endpoint, but I don't get any such url to redirect to.

I use the most recent Kratos `v0.11.1` and Hydra `v2.0.3` and there are no errors displayed in neither logs.

Also, this is how I create the OAuth2 client in Hydra:

```
hydra create client --endpoint http://localhost:4445 --grant-type authorization_code,refresh_token --response-type code --scope openid,offline --redirect-uri http://localhost:3003/callback --audience http://localhost:3003/
```

And how I start the OAuth2 flow:

```
http://localhost:4444/oauth2/auth?response_type=code&client_id=78b11d5d-00f6-4e51-9008-09e819194e87&redirect_uri=http://localhost:3003/callback&state=39f5cea5-7177-4cc1-9f4b-576d4972d6a1&scope=offline
```

Which leads me my login url with only the `login_challenge` query param attached:

```
http://localhost:3002/login?login_challenge=97c23f242ec743ffa3f372fe61c62e25
```

**Comment by vinckr** [comment]
Hello @davidknezic 
This might be a little late, but have you seen this example here: https://github.com/ory/examples/tree/master/kratos-hydra

Also feel free to test it out on [Ory Network](https://console.ory.sh/), there you can avoid the hassle to deploy these services.
